packages/core/src/utils/slang/vendored/slangHelpers.ts

```diff
@@ -0,0 +1,18 @@
+// from https://github.com/NomicFoundation/hardhat-vscode/blob/development/server/src/parser/slangHelpers.ts
+import { getPlatform } from './operatingSystem';
+
+const SUPPORTED_PLATFORMS = [
```

Comment:
I would say though that this seems very wrong to me. Slang should self-report on what platforms it supports to avoid issues when platforms are added or removed.

Reply:
Will suggest this for Slang.

Reply:
Importing Slang in an unsupported platform will actually throw an error, so to support other platforms you should only import it dynamically, which may be somewhat annoying.

You can still use `import type` in the top-level scope of your files though.

At the same time, this could also help you build an alternative `isSlangSupported` function, like this:

```ts
function isSlangSupported() : boolean {
   try {
     require("@nomicfoundation/slang");
   } catch (e) { 
      if ("message" in e && typeof e.message == "string") {
         return e.message === 'Failed to load native binding' || e.message.startsWith('Unsupported architecture') || e.message.startsWith('Unsupported OS');
      }
      
      throw e;
   }
}
```

This was written inside the GitHub UI, so it may not even compile, but it shows the general idea. The error messages are available in `@nomicfoundation/slang/napi-bindings/generated/index.js`. Unfortunately, the generated bindings use a plain `Error`.

Finally, two other notes:

- We've seen virtually no usage of Hardhat in platforms other than the ones listed here, so we strongly believe that it covers most, if not all, the Ethereum ecosystem.
- Once we switch to wasm, this won't be needed anymore :)

Reply:
Thanks for the suggestion, and I agree with you that this should be improved. We are in the process of moving to WASM (in Q3), which will remove the need for platform-specific binaries in any case, so that won't be needed in the future.

In the meantime, the list of supported platforms can be read from the `package.json` `dependencies`, and listed here as well:

https://www.npmjs.com/package/@nomicfoundation/slang?activeTab=dependencies

Reply:
Thanks for the feedback, everyone.  I've changed this to try importing Slang dynamically and if it has an error, we can simply assume the platform is not supported and skip this code path (this way we can avoid checking the error messages since those are not part of the stable API).